python/yellowdog/client/client.py

```diff
@@ -0,0 +1,317 @@
+from armonik.common import TaskDefinition, TaskOptions
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+def run_batch(
+    endpoint: str, partition: str, batch_size: int, iteration: int, scenario: str
+) -> Dict[str, Any]:
+    logger.info(f"Running {scenario} - Iteration {iteration+1}")
+    with grpc.insecure_channel(endpoint) as channel:
+        task_client = ArmoniKTasks(channel)
+        result_client = ArmoniKResults(channel)
+        session_client = ArmoniKSessions(channel)
+        events_client = ArmoniKEvents(channel)
+
+        metrics = {
+            "batch_size": batch_size,
+            "submission_time": 0.0,
+            "processing_time": 0.0,
+            "total_time": 0.0,
+            "scenario": scenario,
+            "iteration": iteration + 1,
+        }
+        start_total = time.time()
+
+        task_options = TaskOptions(
+            max_duration=timedelta(hours=1),
+            max_retries=2,
+            priority=1,
+            partition_id=partition,
+            options={},
+        )
+
+        # Retry logic for session creation
+        max_session_attempts = 3
+        for attempt in range(1, max_session_attempts + 1):
+            try:
+                session_id = session_client.create_session(
+                    task_options, partition_ids=[partition]
+                )
+                logger.info(f"Created session with ID: {session_id}")
+                break
+            except Exception as e:
+                logger.error(f"Session creation attempt {attempt} failed: {e}")
+                if attempt == max_session_attempts:
+                    raise ValueError("Max retries reached for session creation") from e
+                time.sleep(2**attempt)
+
+        task_definitions = []
+        result_ids = []
+        for _ in range(batch_size):
+            results = result_client.create_results_metadata(
+                result_names=["payload", "result"], session_id=session_id
+            )
+            payload_id = results["payload"].result_id
+            result_id = results["result"].result_id
+            result_ids.append(result_id)
+
+            payload = "".encode()
+
+            # Retry logic for payload upload
+            max_upload_attempts = 3
+            for attempt in range(1, max_upload_attempts + 1):
+                try:
+                    result_client.upload_result_data(payload_id, session_id, payload)
+                    break
+                except Exception as e:
+                    logger.error(f"Payload upload attempt {attempt} failed: {e}")
+                    if attempt == max_upload_attempts:
+                        raise ValueError(
+                            "Max retries reached during payload upload"
+                        ) from e
+                    time.sleep(2**attempt)
```

Comment:
Use create_results method that supports creating results and sending payload at the same time.